SKA-ScienceDataProcessor/integration-prototype
# TSK-2762 Add capability to write received data to storage

## Description:

[TSK-2762](https://jira.ska-sdp.org/browse/TSK-2762) Add capability to write received data to storage

Re-arranged receiver code to allow buffers to be locked for writing.
Note this is MVP only, as write performance is not yet good enough. Also
currently needs to run as root inside container to avoid FS permission
problems with bind mounts.

## Testing instructions:

From the SIP repository root:
```bash
cd sip/science_pipeline_workflows/ingest_visibilities/
```

To deploy the services (execute this command just past the minute or half-minute mark, as the senders will start based on the current time; 30 seconds should be long enough for everything to start up):
```bash
docker stack deploy -c docker-compose-p3-c-multi-sender.yml i
```

To inspect the logs:
```bash
docker service logs -t i_recv
```

To stop everything:
```bash
docker stack rm i
```

To remove the files written to storage after the test:
```bash
docker run --rm -it --mount "type=bind,src=/mnt/beegfs,dst=/app/output" ubuntu /bin/bash -c "rm /app/output/ingest_run*"
```

## Types of changes
